flake.nix

```diff
@@ -0,0 +1,63 @@
+{
+  description = "A Nix-flake-based development environment for Cursorless";
+
+  inputs.nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+
+  outputs =
+    { self, nixpkgs }:
+    let
+      supportedSystems = [
+        "x86_64-linux"
+        "aarch64-linux"
+        "x86_64-darwin"
+        "aarch64-darwin"
+      ];
+      forEachSupportedSystem =
+        f: nixpkgs.lib.genAttrs supportedSystems (system: f { pkgs = import nixpkgs { inherit system; }; });
+      pythonVersion = builtins.replaceStrings [ "py" ] [
+        "python"
+      ] (nixpkgs.lib.importTOML ./pyproject.toml).tool.ruff.target-version;
+    in
+    {
+      devShells = forEachSupportedSystem (
+        { pkgs }:
+        {
+          default = pkgs.mkShell {
+            packages =
+              let
+                python = pkgs.${pythonVersion};
+                pythonPackages = pkgs."${pythonVersion}Packages";
+              in
+              [
+                pkgs.corepack
+                pkgs.vsce
+                pkgs.pre-commit
+
+                python
+                # Dependencies from pre-commit python scripts
+                # run `pre-commit run --all-files` to find out what you need.
+                pythonPackages.identify
+                pythonPackages.cfgv
```

Comment:
Shouldn't these be direct dependencies of pre-commit, and used only by pre-commit itself?

Reply:
In theory, but in the past I've run into plenty of pre-commit hooks that don't actually install all dependencies properly, which means you end up needing to manually install. In this case when I run `pre-commit run --all-files` python complains the ones hardcoded above don't exist... so seems like a bug in their hook. If I want to be able to run pre-commit, seems I need to do it for now.

example:
```bash
❯ pre-commit run --all-files
Check hooks apply to the repository......................................Failed
- hook id: check-hooks-apply
- exit code: 1

Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/nix/store/f87ha48qbkhfz6h2yg2dxp6s1kcbqdmp-pre-commit-3.6.2/lib/python3.11/site-packages/pre_commit/meta_hooks/check_hooks_apply.py", line 7, in <module>
    from pre_commit import git
  File "/nix/store/f87ha48qbkhfz6h2yg2dxp6s1kcbqdmp-pre-commit-3.6.2/lib/python3.11/site-packages/pre_commit/git.py", line 9, in <module>
    from pre_commit.util import CalledProcessError
  File "/nix/store/f87ha48qbkhfz6h2yg2dxp6s1kcbqdmp-pre-commit-3.6.2/lib/python3.11/site-packages/pre_commit/util.py", line 16, in <module>
    from pre_commit import parse_shebang
  File "/nix/store/f87ha48qbkhfz6h2yg2dxp6s1kcbqdmp-pre-commit-3.6.2/lib/python3.11/site-packages/pre_commit/parse_shebang.py", line 7, in <module>
    from identify.identify import parse_shebang_from_file
ModuleNotFoundError: No module named 'identify'
```

Reply:
That traceback points to pre-commit itself… sounds like the pre-commit package is straight up broken.

Reply:
I just tested to update the flake.lock and run pre-commit 3.7.1 and it's the same. 

But ya you are right, it's a bug in the nix package: https://github.com/NixOS/nixpkgs/issues/270805

Reply:
I'll subscribe and see if I can move it forward, but in the meantime will just keep the dependencies.

Reply:
Ok I lied, I updated it to use a workaround mentioned in are related PR https://github.com/NixOS/nixpkgs/pull/251418. It's cleaner than adding the individual dependencies, although apparently not necessarily unbreakable. Seems to work so far though.